File: editor/src/de/project/ice/editor/undoredo/AddEntityAction.java
package de.project.ice.editor.undoredo;

import com.badlogic.ashley.core.Engine;
import com.badlogic.ashley.core.Entity;

public class AddEntityAction implements UndoableRedoableAction
{
    private Entity entity;
    private Engine engine;

    public AddEntityAction(Entity entity, Engine engine)
    {
        this.entity = entity;
        this.engine = engine;
    }


    @Override
    public void undo()
    {
        engine.removeEntity(entity);
    }

    @Override
    public void redo()
    {
        engine.addEntity(entity);
    }

}

File: editor/src/de/project/ice/editor/undoredo/AddEntityAction.kt
package de.project.ice.editor.undoredo

import com.badlogic.ashley.core.Engine
import com.badlogic.ashley.core.Entity

class AddEntityAction(private val entity: Entity, private val engine: Engine) : UndoableRedoableAction {


    override fun undo() {
        engine.removeEntity(entity)
    }

    override fun redo() {
        engine.addEntity(entity)
    }

}
